test/functional/terminal/tui_spec.lua

```diff
@@ -2171,7 +2180,7 @@ describe('TUI FocusGained/FocusLost', function()
       msg3                                              |
       msg4                                              |
       msg5                                              |
-      {10:Press ENTER or type command to continue}{1: }          |
+      {2:Press ENTER or type command to continue}{1: }          |
```

Comment:
With this PR, a terminal in Neovim reports `foreground` and `background` colors to a nested Neovim instance.

Under tests, the nested Neovim instance sets its `background` to `light` after receiving information from the outer Neovim terminal that it's running in. Prior to this PR, the `background` would have been kept as `dark`, since the OSC background query would not have been responded to. With the `background` set to `light` instead of `dark`, the color of "Press ENTER or type command to continue" changes.

The outer Neovim instance seems to have its `background` set to `light` independent of the background color of the terminal I use. I'm not sure where in the tests that this is set.

Reply:
@dstein64 could you merge the current master here? It requires to revolve some conflicts.